### ISSUE TITLE
Select datasets for fine-tuning

### ISSUE THREAD

**Issue by jack89roberts** [alan-turing-institute/ARC-LoCoMoSeT#3]
### Criteria

- No licensing or ethical concerns
- Sufficiently difficult to see differences between the performance of different models
- _Ideally:_ Variety in
  - No. of classes
  - No. of samples
  - Domain and "similarity" to ImageNet
  - Difficulty
- _Ideally:_
  - Well-understood/used elsewhere
  - Available as a HuggingFace or torchvision dataset

### Initial Candidates (statement of work)

- Beans: https://huggingface.co/datasets/beans
- Oxford Pets: https://huggingface.co/datasets/pcuenq/oxford-pets
- ~~Human Activity Recognition: https://huggingface.co/datasets/Bingsu/Human_Action_Recognition~~
  - Includes faces and images with watermarks - probably better to avoid

**Comment by jack89roberts** [comment]
- ObjectNet:
  - https://objectnet.dev/
  - https://paperswithcode.com/sota/image-classification-on-objectnet
  - State of the art: 83% accuracy
  - Some overlapping classes with ImageNet (~100 out of 300 classes) but designed to be difficult (different angles, backgrounds etc..)

**Comment by jack89roberts** [comment]
| Dataset | No. Classes | No. Images | SotA % Accuracy | On HuggingFace | Notes |
| -------- | ----------- | ------------- | --------------- | ---------------- | ------- |
| [ibean](https://github.com/AI-Lab-Makerere/ibean/) | 3 | 1300 | [>97.7?](https://huggingface.co/nateraw/vit-base-beans) | [Yes](https://huggingface.co/datasets/beans) | |
| [Oxford-IIIT Pet](https://www.robots.ox.ac.uk/~vgg/data/pets/) | 37 | 7349 | [96.7](https://paperswithcode.com/sota/fine-grained-image-classification-on-oxford-1) | [Yes](https://huggingface.co/datasets/pcuenq/oxford-pets) | |
| [ObjectNet](https://objectnet.dev/) | 313 | 50,000 | [82.7](https://paperswithcode.com/sota/image-classification-on-objectnet) | No | 113 classes overlap ImageNet, intended as test set only |
| [BigEarthNet](https://bigearth.net/) | 19 | 590,326 | [91.8](https://paperswithcode.com/sota/multi-label-image-classification-on) | No | Multilabel classification (SotA metric is mAP), up to 43 classes available |
| [Food-101](https://data.vision.ee.ethz.ch/cvl/datasets_extra/food-101/) | 101 | 101,000 | [96.2](https://paperswithcode.com/sota/fine-grained-image-classification-on-food-101) | [Yes](https://huggingface.co/datasets/food101) | |
| [Describable Textures](https://www.robots.ox.ac.uk/~vgg/data/dtd/index.html) | 47 | 5640 | [84.0](https://paperswithcode.com/sota/image-classification-on-dtd) | [Maybe](https://huggingface.co/datasets/cansa/Describable-Textures-Dataset-DTD), or [Torchvision](https://pytorch.org/vision/stable/datasets.html) | |
| [Places 205](http://places.csail.mit.edu/index.html) | 205 | 2,500,000 | [71.7](https://paperswithcode.com/sota/image-classification-on-places205) | No, [Torchvision](https://pytorch.org/vision/stable/generated/torchvision.datasets.Places365.html#torchvision.datasets.Places365) has Places365 | |
| [iNaturalist 2019](https://github.com/visipedia/inat_comp/tree/master/2019) | 1,010 | 268,243 | [88.5](https://paperswithcode.com/sota/image-classification-on-inaturalist-2019) | [Butterflies subset](https://huggingface.co/datasets/huggan/inat_butterflies), or [Torchvision](https://pytorch.org/vision/stable/generated/torchvision.datasets.INaturalist.html#torchvision.datasets.INaturalist) | Other years available too |
| [Human Activity Recognition](https://aiplanet.com/challenges/233/data-sprint-76-human-activity-recognition-233/overview/about) | 15 | 18,000 | [>84.8?](https://aiplanet.com/challenges/233/data-sprint-76-human-activity-recognition-233/leaderboard/practice) | [Yes](https://huggingface.co/datasets/Bingsu/Human_Action_Recognition) | Watermarks, faces |
| [Caltech Birds](http://www.vision.caltech.edu/datasets/cub_200_2011/) | 200 | 11,788 | [93.1](https://paperswithcode.com/sota/fine-grained-image-classification-on-cub-200) | [Maybe](https://huggingface.co/datasets/student/CUB_birds_200_2011) | Images overlap with ImageNet |
| [Stanford Dogs](http://vision.stanford.edu/aditya86/ImageNetDogs/) | 120 | 20,580 | [93.6](https://paperswithcode.com/sota/fine-grained-image-classification-on-stanford-1) | [Yes](https://huggingface.co/datasets/ksaml/Stanford_dogs) (question mark on labels) | Images overlap with ImageNet |
| [NABirds](https://dl.allaboutbirds.org/nabirds) | ~550 | 48,000 | [93.0](https://paperswithcode.com/sota/fine-grained-image-classification-on-nabirds) | [Maybe](https://huggingface.co/datasets/anjunhu/naively_captioned_nabirds) (annotations rather than labels) |  |
| [Caltech101](https://data.caltech.edu/records/mzrjq-6wc02) | 101 | ~9000 | [98.0](https://paperswithcode.com/sota/fine-grained-image-classification-on-caltech) | Maybe, or [torchvision](https://pytorch.org/vision/stable/generated/torchvision.datasets.Caltech101.html#torchvision.datasets.Caltech101) |  |
| [CIFAR-10](https://www.cs.toronto.edu/~kriz/cifar.html) | 10 | 60000 | [99.7](https://paperswithcode.com/sota/image-classification-on-cifar-10) |  [Yes](https://huggingface.co/datasets/cifar10) |  |


Other [Torchvision datasets](https://pytorch.org/vision/stable/datasets.html) may be worth a look (e.g. Caltech 256, German Traffic Sign Recognition Benchmark (GTSRB), Country 211, FGVC aircraft).

Something completely different would be nice to have, e.g. document classification e.g. [RVL CDIP](https://huggingface.co/datasets/aharley/rvl_cdip), but that's a greyscale dataset.

**Comment by jack89roberts** [comment]
Thoughts:

- Out of the 3 datasets in the SoW beans is maybe too easy, and Human Activity Recognition seems a bit questionable from a licensing standpoint. So I propose to start with **Oxford-IIIT Pet**, probably as the _easiest_ dataset we try.
- After that, we evaluate and try a harder dataset. Not so many of these on HuggingFace (that I could find benchmarks for), but
  - Out of the Torchvision ones above Describable Textures, Places 365, and iNaturalist are options in SoTA < 90%. 
  - ObjectNet is interesting as something that partially overlaps with ImageNet but is intentionally more difficult
  - BigEarthNet is most different to ImageNet and likely more difficult than SoTA suggests. But may take too long to train/get working for the project.

**Comment by jack89roberts** [comment]
- Open images: https://storage.googleapis.com/openimages/web/index.html , maybe if we want to curate something specific (harder to find SOTA to compare to)

**Comment by jack89roberts** [comment]
Few other HuggingFace ones that may be worth a look/see if there are SOTA results we can compare to:

- [wikiart](https://huggingface.co/datasets/huggan/wikiart) - classify artist from painting
- [rvl_cdip](https://huggingface.co/datasets/aharley/rvl_cdip) - classify document type
- [skincancer](https://huggingface.co/datasets/marmal88/skin_cancer)
- [ACL-fig](https://huggingface.co/datasets/citeseerx/ACL-fig) - classify type of figure from papers
- [ecoset](https://huggingface.co/datasets/kietzmannlab/ecoset)
- [svhn](https://huggingface.co/datasets/svhn) - streetview house numbers
- [quickdraw](https://huggingface.co/datasets/quickdraw) - hand (computer?) drawn sketches
- [Chest X-rays](https://huggingface.co/datasets/keremberke/chest-xray-classification)

**Comment by jack89roberts** [comment]
- [imagenet-hard](https://huggingface.co/datasets/taesiri/imagenet-hard), ~same classes as ImageNet-1k (I think) but from a selection of much more difficult datasets: ImageNet, ImageNet-ReaL, ImageNet-V2, ImageNet-A, ImageNet-C, ImageNet-R, ImageNet-Sketch, and ObjectNet. SOTA 39% accuracy (but that's if using it as a test set rather than a training set I think).

**Comment by eddableheath** [comment]
Can we close this?
